src/test/java/uk/gov/companieshouse/api/testdata/service/impl/IdentityServiceImplTest.java

```diff
@@ -0,0 +1,130 @@
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.util.Optional;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import uk.gov.companieshouse.api.testdata.model.entity.Identity;
+import uk.gov.companieshouse.api.testdata.model.rest.IdentityData;
+import uk.gov.companieshouse.api.testdata.model.rest.IdentitySpec;
+import uk.gov.companieshouse.api.testdata.repository.IdentityRepository;
+import uk.gov.companieshouse.api.testdata.service.RandomService;
+
+
+@ExtendWith(MockitoExtension.class)
+class IdentityServiceImplTest {
+
+    @Mock
+    private IdentityRepository identityRepository;
+
+    @Mock
+    private RandomService randomService;
+
+    @InjectMocks
+    private IdentityServiceImpl identityServiceImpl;
+
+    @Test
+    void testCreateIdentity() {
+        IdentitySpec identitySpec = new IdentitySpec();
+        identitySpec.setUserId("randomised");
+        identitySpec.setEmail("test@test.com");
+        identitySpec.setVerificationSource("source");
+        String generatedIdentityId = "randomised";
+        when(randomService.getString(24)).thenReturn(generatedIdentityId);
+        final var createdDate = LocalDateTime.now(ZoneId.of("UTC")).toInstant(ZoneOffset.UTC);
+        IdentityData createdIdentity = identityServiceImpl.create(identitySpec);
+        assertEquals(createdIdentity.getId(), generatedIdentityId,
+                "ID should match the generated ID");
+
+        ArgumentCaptor<Identity> identityCaptor = ArgumentCaptor.forClass(Identity.class);
+        verify(identityRepository).save(identityCaptor.capture());
+        Identity savedIdentity = identityCaptor.getValue();
+
+        assertEquals(identitySpec.getEmail(), savedIdentity.getEmail(), "Email should match");
+        assertEquals(identitySpec.getUserId(), savedIdentity.getUserId(), "User ID should match");
+        assertEquals(identitySpec.getVerificationSource(), savedIdentity.getVerificationSource(),
+                "Verification source should match");
+        assertEquals(generatedIdentityId, savedIdentity.getId(),
+                "ID should match the generated ID");
+        assertEquals("VALID", savedIdentity.getStatus(), "Status should be VALID");
+        assertTrue(createdDate.isBefore(savedIdentity.getCreated()),
```

Comment:
We could check the exact date (rather than "is before") if we created a protected method in the service to get today's date and we used an spy to make it return the value we expect from the test